resources/file.go

```diff
@@ -102,6 +108,20 @@ func (obj *FileRes) Validate() error {
 		return fmt.Errorf("Can't specify Content when creating a Dir.")
 	}
 
+	if obj.Mode != "" {
+		if _, err := obj.mode(); err != nil {
+			return err
```

Comment:
if it would provide useful context, we always like to errwrap :) -- whether it makes sense really depends on what mode() does.